crates/burn-tensor/src/tensor/ops/int_tensor.rs

```diff
@@ -1168,4 +1167,35 @@ pub trait IntTensorOps<B: Backend> {
 
     /// Bitwise right shift operation for Int Tensors with a scalar
     fn bitwise_right_shift_scalar(lhs: IntTensor<B>, rhs: IntElem<B>) -> IntTensor<B>;
+
+    /// Applies the matrix multiplication operation.
+    ///
+    /// `C = AB`
+    ///
+    /// # Panics
+    ///
+    /// If the two tensors don't have a compatible shape.
+    fn matmul(lhs: IntTensor<B>, rhs: IntTensor<B>) -> IntTensor<B> {
+        let mut lhs_shape: Vec<usize> = lhs.shape().dims.clone();
+        lhs_shape.push(1);
+        let lhs_shape: Shape = Shape::from(lhs_shape);
+        let lhs = B::int_reshape(lhs, lhs_shape);
+
+        let mut rhs_shape: Vec<usize> = rhs.shape().dims.clone();
+        rhs_shape.insert(rhs_shape.len() - 2, 1);
+        let rhs_shape: Shape = Shape::from(rhs_shape);
```

Comment:
I would not provide a default implementation here

Reply:
would it be in numeric? how do I juggle the situation with float?

Reply:
Moved